### ISSUE TITLE
Introduce ConstAstVisitor

### ISSUE THREAD

**Issue by tristan0x** [BlueBrain/nmodl#293]
An Ast visitor which is not supposed to modify the visited ast should not be able to do it. It provides safety inside the visitor implementation and closure when using it.

This is why I suggest adding a `ConstAstVisitor` class that can only visit const nodes.

The Ast nodes would have an additional `accept` method:
```c++
  /** \name Visitors entry point.
     ** \{ */
    /// Accept a const visitor \a v.
    void accept(ConstAstVisitor& v) const override;
    /// Accept a non-const visitor \a v.
    void accept(AstVisitor& v) override;
    /** \} */
```

**Comment by ohm314** [comment]
I like your suggestion and I think this would be an easy improvement for a free-friday.

**Comment by cattabiani** [comment]
+1 for me too

**Comment by tristan0x** [comment]
Fixed in #427
